src/Orleans.Runtime/Catalog/Catalog.cs

```diff
@@ -283,7 +283,7 @@ public void UnregisterSystemTarget(ISystemTarget target)
             }
 
             // Initialize the new activation asynchronously.
-            var cancellation = new CancellationTokenSource(collectionOptions.Value.ActivationTimeout);
+            using var cancellation = new CancellationTokenSource(collectionOptions.Value.ActivationTimeout);
```

Comment:
This is erroneous, unfortunately - to implement this correctly (with disposal) will require a re-think on our part. Could you log an issue for it and revert this change?

Reply:
#9082 is created